frontend/components/forms/fields/AutocompleteDropdown/AutocompleteDropdown.tsx

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import { Async, OnChangeHandler, Option } from "react-select";
+import classnames from "classnames";
+
+// ignore TS error for now until these are rewritten in ts.
+// @ts-ignore
+import local from "utilities/local";
+// @ts-ignore
+import debounce from "utilities/debounce";
+
+const baseClass = "autocomplete-dropdown";
+
+interface IAutocompleteDropdown {
+  id: string;
+  placeholder: string;
+  onChange: OnChangeHandler;
+  resourceUrl: string;
+  valueKey: string;
+  labelKey: string;
+  value: Option[];
+  disabled?: boolean;
+  optionComponent?: JSX.Element;
+  className?: string;
+}
+
+const debounceOptions = {
+  timeout: 300,
+  leading: false,
+  trailing: true,
+};
+
+const createUrl = (baseUrl: string, input: string) => {
+  return `/api${baseUrl}?query=${input}`;
+};
+
+const AutocompleteDropdown = (props: IAutocompleteDropdown): JSX.Element => {
+  const {
+    className,
+    disabled,
+    placeholder,
+    onChange,
+    id,
+    resourceUrl,
+    valueKey,
+    labelKey,
+    value,
+  } = props;
+
+  const wrapperClass = classnames(baseClass, className);
+
+  const getOptions = debounce((input: string) => {
+    if (!input) {
+      return Promise.resolve({ options: [] });
+    }
+
+    return fetch(createUrl(resourceUrl, input), {
+      headers: {
+        authorization: `Bearer ${local.getItem("auth_token")}`,
+      },
+    })
+      .then((res) => {
+        return res.json();
+      })
+      .then((json) => {
+        return { options: json.users };
+      })
+      .catch((err) => {
+        console.log("There was an error", err);
+      });
```

Comment:
Can we render this error on the page?

Reply:
yes will need to talk to @noahtalerman about what errors look like for this